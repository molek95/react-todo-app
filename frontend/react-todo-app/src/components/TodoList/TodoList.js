import React, {useState, useEffect} from 'react'
import TodoForm from '../TodoForm/TodoForm'
import Todo from '../Todo/Todo'

function TodoList() {
    const [todos, setTodos] = useState([]);

    useEffect(() => {
        fetch('http://localhost:3000/api/tasks')
            .then(response => response.json())
            .then(data => {
                setTodos(Object.values(data))
            });
    }, []);

    const addTodo = todo => {
        if(!todo.text) {
            return
        }
        
        const requestOptions = {
            method: 'POST',
            headers: {'Content-Type': 'application/json'},
            body: JSON.stringify(todo)
        };
        fetch('http://localhost:3000/api/tasks', requestOptions)
            .then(() => {
                const newTodos = [todo, ...todos];
                setTodos(newTodos);
            })
    }

    const completeTodo = id => {
        let updatedTodos = todos.map(todo => {
            if (todo.id === id) {
                todo.isComplete = !todo.isComplete
            }
            return todo
        })
        setTodos(updatedTodos);
    }

    const removeTodo = id => {
        fetch('http://localhost:3000/api/tasks/' + `${id}`, {method: 'DELETE'})
            .then(() => {
                const removeArr = [...todos].filter(todo => todo.id !== id)
                setTodos(removeArr);
            })
    }

    const editTodo = (id, currentValue) => {
        if(!currentValue.text) {
            return
        }
        const requestOptions = {
            method: 'PUT',
            headers: {'Content-Type': 'application/json'},
            body: JSON.stringify(currentValue)
        }
        fetch('http://localhost:3000/api/tasks/' + `${id}`, requestOptions)
            .then(() => {
                setTodos(prev => prev.map(item => (item.id === id ? currentValue : item)));
            })
    }
    
    return (
        <div>
            <h1>React TODO app</h1>
            <TodoForm onSubmit={addTodo} />
            <Todo todos={todos} completeTodo={completeTodo} removeTodo={removeTodo} editTodo={editTodo}/>
        </div>
    )
}

export default TodoList
