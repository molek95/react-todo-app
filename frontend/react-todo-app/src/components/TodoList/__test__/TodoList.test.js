import React from 'react';
import { shallow } from 'enzyme';
import TodoList from './../TodoList';

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

Enzyme.configure({ adapter: new Adapter() });

describe('<TodoList /> with no props', () => {
    const todoList = shallow(<TodoList />);

    it('should match the snapshot', () => {
        expect(todoList.html()).toMatchSnapshot();
    });

})
