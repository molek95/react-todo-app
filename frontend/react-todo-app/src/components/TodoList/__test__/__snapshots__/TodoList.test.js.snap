// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> with no props should match the snapshot 1`] = `"<div><h1>React TODO app</h1><form class=\\"todo-form\\"><input type=\\"text\\" placeholder=\\"Add a todo\\" value=\\"\\" name=\\"text\\" class=\\"todo-input\\"/><button class=\\"todo-button\\">Add todo</button></form><div></div></div>"`;
